internal/replication/blockreplicator_3node_test.go

```diff
@@ -0,0 +1,313 @@
+package replication_test
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/stretchr/testify/require"
+)
+
+// Scenario:
+// - Start 3 nodes together, wait for leader,
+// - Stop the leader, wait for new leader,
+// - Stop the leader, remaining node has no leader.
+func TestBlockReplicator_3Node_StartCloseStep(t *testing.T) {
+	env := createClusterEnv(t, "info", 3)
+	require.Equal(t, 3, len(env.nodes))
+	defer os.RemoveAll(env.testDir)
+
+	// start 3 at once
+	for _, node := range env.nodes {
+		err := node.Start()
+		require.NoError(t, err)
+	}
+
+	// wait for an agreed leader
+	sameLeaderCond := func() bool {
+		return env.AgreedLeaderIndex() >= 0
+	}
+	assert.Eventually(t, sameLeaderCond, 30*time.Second, 100*time.Millisecond)
+
+	//close the leader, wait for some node to become a new leader
+	leaderIndex1 := env.FindLeaderIndex()
+	require.True(t, leaderIndex1 >= 0)
+	t.Logf("Leader #1 index: %d", leaderIndex1)
+	err := env.nodes[leaderIndex1].Close()
+	require.NoError(t, err)
+	isLeaderCond := func() bool {
+		return env.FindLeaderIndex() >= 0
+	}
+	assert.Eventually(t, isLeaderCond, 30*time.Second, 100*time.Millisecond)
+
```

Comment:
we can also check whether the new leader is different from the old (closed node).

Reply:
good idea.